### DISCUSSION TITLE
X98 Mini - S905W2

### DISCUSSION THREAD

**Discussion by dolaru** [devmfc/debian-on-amlogic#105] [Q&A]
Hello 👋🏽. I've got my hands on an X98 mini box and I'm trying to get a mainline kernel running on it, but there doesn't seem to be a suitable DTB around.

Snooping around the device tree in the Android distribution that came on it I found references to `S4` and `AP201`. and there is a device tree definition for `AP201` [here](https://github.com/CoreELEC/u-boot/blob/3efc85a8370796bcec3bcadcdecec9aed973f4a9/arch/arm/dts/meson-s4-ap201.dts).

The mainline kernel seems to have the `S4` [base](https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/amlogic/meson-s4.dtsi).

Any advice on what the next steps would be to get a suitable device tree image built for this box? 

## Specs
CPU: Amlogic S905W2 quad-core ARM Cortex A35 up to 1.8GHz
GPU: Mali-G31 MP2 OpenGL ES 1.1/2.0/3.2, OpenCL 2.0, Vulkan 1.1
Memory: DDR3 4GB, EMMC 32GB
Network: 100M Ethernet, Dual Wi-Fi 2.4 & 5G

### Factory OS
Android 11

### I/O
1 * HDMI 4K 2K UHD output, HD 2.0A
1 * AV OUT 480i/576i standard definition output
2 * USB 2.0 * USB host port, 1 * USB device
1 * IR remote receiver connection connector
1 * RJ45 Ethernet interface
1 * TF card Support 4GB/8GB/16GB/32GB/64GB

Hardware support decoding: AV1, VP9, ​​H.265 4Kx2K@60fps HDR10+, HDTV 2.1

## Board
![B8A4E13C-D7B5-42B3-9E2F-0C5CB05B7B00_1_102_a](https://github.com/user-attachments/assets/828f171b-b850-4b32-8e01-fa59104bd0b8)

**Comment by ProfBoc75** [comment]
@dolaru : I have exactly the same board "BM844_141_V2.3_21305" but 2G/16GB model. "X98 mini" model. My sticker in the middle is "2+16+W522A" then my batch label is "ZF20220830"

I tested successfully with last version "Devmfc_Debian-Bookworm_6.12.11-meson64_Minimal-25.01.24" with the "x96miniw2" or the "s905w2_generic" in the boot.config file.

I didn't test every thing, not yet try to install it onto the emmc.
